bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -0,0 +1,177 @@
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO.Device;
+import org.openhab.core.i18n.LocaleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link MeaterRestAPI} class defines the Meater REST API
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterRestAPI {
+    private static final String API_ENDPOINT = "https://public-api.cloud.meater.com/v1/";
+    private static final String JSON_CONTENT_TYPE = "application/json";
+    private static final String LOGIN = "login";
+    private static final String DEVICES = "devices";
+    private static final int MAX_RETRIES = 3;
+
+    private final Logger logger = LoggerFactory.getLogger(MeaterRestAPI.class);
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private final MeaterBridgeConfiguration configuration;
+    private String authToken = "";
+    private String userId = "";
+    private LocaleProvider localeProvider;
+
+    public MeaterRestAPI(MeaterBridgeConfiguration configuration, Gson gson, HttpClient httpClient,
+            LocaleProvider localeProvider) {
+        this.gson = gson;
+        this.configuration = configuration;
+        this.httpClient = httpClient;
+        this.localeProvider = localeProvider;
+    }
+
+    public boolean refresh(Map<String, MeaterProbeDTO.Device> meaterProbeThings) {
+        try {
+            // Login
+            login();
+
+            MeaterProbeDTO dto = getDevices(MeaterProbeDTO.class);
+            List<Device> devices = dto.getData().getDevices();
+
+            if (devices != null) {
+                if (!devices.isEmpty()) {
+                    for (Device meaterProbe : devices) {
+                        meaterProbeThings.put(meaterProbe.id, meaterProbe);
+                    }
+                } else {
+                    meaterProbeThings.clear();
+                }
+            }
+            return true;
+        } catch (MeaterException | JsonSyntaxException e) {
+            logger.warn("Failed to refresh! {}", e.getMessage());
+        }
+        return false;
+    }
+
+    private void login() throws MeaterException {
+        try {
+            // Login
+            String json = "{ \"email\": \"" + configuration.email + "\",  \"password\": \"" + configuration.password
+                    + "\" }";
+            Request request = httpClient.newRequest(API_ENDPOINT + LOGIN).method(HttpMethod.POST);
+            request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+            request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+            request.content(new StringContentProvider(json), JSON_CONTENT_TYPE);
+
+            logger.trace("HTTP POST Request {}.", request.toString());
```

Comment:
This will be logged like this:
`HTTP POST Request HttpRequest[POST /v1/login HTTP/1.1]@c9a9ad.`

So maybe it would be worth some custom formatting, or alternatively stripping the "HTTP POST Request " as it seems redundant.

Reply:
Fixed.